ui/client/components/graph/ExpressionSuggester.js

```diff
@@ -35,20 +35,21 @@ export default class ExpressionSuggester {
     const variableNames = _.keys(variables)
     const variableAlreadySelected = _.some(variableNames, (variable) => { return _.includes(value, `${variable}.`) })
     const variableNotSelected = _.some(variableNames, (variable) => { return _.startsWith(variable.toLowerCase(), value.toLowerCase()) })
+    let result;
     if (variableAlreadySelected && focusedClazz) {
       const currentType = this._getTypeInfo(focusedClazz)
       const inputValue = this._justTypedProperty(value)
       const allowedMethodList = this._getAllowedMethods(currentType)
-      return inputValue.length === 0 ? allowedMethodList : this._filterSuggestionsForInput(allowedMethodList, inputValue)
+      result = inputValue.length === 0 ? allowedMethodList : this._filterSuggestionsForInput(allowedMethodList, inputValue)
     } else if (variableNotSelected && !_.isEmpty(value)) {
       const allVariablesWithClazzRefs = _.map(variables, (val, key) => {
         return {'methodName': key, 'refClazz': val}
       })
-      return this._filterSuggestionsForInput(allVariablesWithClazzRefs, value)
-    }
-    else {
-      return []
+      result = this._filterSuggestionsForInput(allVariablesWithClazzRefs, value)
+    } else {
+      result = []
```

Comment:
You can do  at start ` let result = []` and remove this else.  Or better in each if do simply return and at end return [];

Reply:
I can't do return in each if/else because need to wrap return in `Promise` but I can init variable with empty array :+1: